_posts/2021-08-16-Day-6-Post.md

```diff
@@ -0,0 +1,59 @@
+author: Kevin Fong
+---
+## C++ as a Programming Language
+
+C++ is a general purpose programming language created as an extension of the C programming language. It was designed with an orientation toward system programming and embedded, resource-contained software and large systems, with performance, efficiency, and flexibility of us as its design highlights.
+
+C++ is an object oriented, staticly typed, compiled language. As a static typed language, all objects have their data type declared upon definition. As a compiled language, the source code is translated using compilers to produce machine code. Once compiled, the machine code is often undecipherable to humans. Because the compilation is performed in one step, compilation errors will prevent the code from compiling, and the code must debugged before the code is run and executed.
+
+## Comparing Python and C++"
+
+Python by comparison is a high-level, mutli-paradigm, dynamically typed, garbage-collected, interpreted language. As an interpreted language, Python code is not compiled into machine instructions and is ready to be read and interpreted as is. Because it is dynamically written, data types are not declared, making Python more flexible and easy to use. However C++ is able to perform more effciently because all data types are declared at the point of compilation, so the machine does not have to consume any extra time to interpret the code to determine what data types are being used.
+
+In terms of syntax, C++ is a more rigoursly structured language. As all object need to be declared, and syntax accuracy is more important, C++ may use more lines of code to accomplish the same obective in Python. On the other hand, as a lanugage Python is sensitive to whitespace, meaning code written in python uses indents to separate sections and layer of code. C++ ignores whitespace, and uses curly braces `{}` to delineate sections of code and `;` to seperate lines. Between the two, only Python forces the programmer to organize the code visually through indents, although it is good practice to or intent in C++ to make the code more legible.
+
+C++ also uses various shortcuts or conventions which I feel add some character to the language. As in the namesake, C++ has an increment operator (`++`) as well as a decrement operator (`--`). C++ will also use the conventions of `||` and `&&` for **or** and **and** respectively.
+
+## Debugging C++
+
+Because C++ is compiled lanugage, the code must be free of errors for the code to compile successfully. C++ has a strict syntax, thus the compilers are able to idenify any and all deviations in the code from expected syntax that will prevent the compilers from interpretting to code as we expect. As C++ will try to give full details and insight about each instance of error it encounters, it often produces large returns of error messages that become ironically increasingly less useful to the human programmer to troubleshoot and fix.
```

Comment:
```suggestion
Because C++ is compiled language, the code must be free of errors for the code to compile successfully. C++ has a strict syntax, thus the compilers are able to identify any and all deviations in the code from expected syntax that will prevent the compilers from interpreting to code as we expect. As C++ will try to give full details and insight about each instance of error it encounters, it often produces large returns of error messages that become ironically increasingly less useful to the human programmer to troubleshoot and fix.
```